task_1/main.cpp

```diff
@@ -0,0 +1,211 @@
+	
+	StringListRemoveDuplicates(list1);
+	StringListPlot(list1);
+	
+	StringListRemove(list1, str2);
+	StringListPlot(list1);
+
+	StringListRemove(list1, str11);
+	StringListPlot(list1);
+
+	StringListRemove(list1, str3);
+	StringListPlot(list1);
+	
+	StringListRemove(list1, str2);
+	StringListPlot(list1);
+
+	StringListRemove(list1, str11);
+	StringListPlot(list1);
+
+	StringListRemove(list1, str3);
+	StringListPlot(list1);
+
+	StringListRemoveDuplicates(list1);
+	StringListPlot(list1);
+
+	printf ("FIRST INDEX IS %i OF %s", StringListIndexOf(list1, str11), str11);
+	printf ("FIRST INDEX IS %i OF %s", StringListIndexOf(list1, str12), str12);
+	printf ("FIRST INDEX IS %i OF %s", StringListIndexOf(list1, str3), str3);
+
+	printf ("SIZE:%i\n", StringListSize(list1));
+	StringListDestroy(&list1);
+	printf ("list1:%p\n", list1);
+
+	free(str1);	free(str2);	free(str3);	free(str11);	free(str12);
+	return 0;
+}
+
+void StringListInit(char*** list){
+	printf ("-->Init<--");
+	if (!list)
+		return;
+	if ( !(* list = (char**)malloc (2 * sizeof(char**))) )
+		return;
+	printf ("\tallocated\t%p\n", * list);
+	** list = 0;					//	a
+	*(++(* list)) = 0;				//	b
+}
+void StringListDestroy(char*** list) {/* Destroy list and set pointer to NULL. */
+	printf ("->destroy<-\t");
+	if (!list)
+		return;
+	--(*list);
+	while (**list != 0) {
+		char ** nextItem = (char **)**list;
+		++nextItem;
+		nextItem = (char **)* nextItem;
+		printf ("\n\t\titem:\t\t%p", **list);
+		free(**list);
+		** list = (char *)nextItem;
+	}
+	printf ("\n\t\thead:\t\t%p\n", *list);
+	free(*list);
+	*list = 0;
+}
+
+void StringListAdd(char** list, char * str) {/* Inserts value at the end of the list. */
+	printf ("-->adding<--\t%p\t%s", str, str);
+	if (!list)
+		return;
+	char ** item;
+	if ( !(item = (char**)malloc (2 * sizeof(char**))) )
+		return;
+	* item = str;					//	a
+	printf ("\t\tallocated\t%p\n", item);
+
+	if (*list == 0) {				//	first addition
+		* list = (char *)item;			//	b
+		--list;
+		* list = (char *)item;			//	c
+		*(++item) = 0; 				//	d
+	} else {					//	next additions
+		char** tmp = (char**)*list;		//	b1
+		++tmp;					//	b1
+		* tmp = (char *)item;			//	b1
+		* list = (char *)item;			//	c1
+		*(++item) = 0;				//	d1
+	}
+}
+
+void StringListPlot(char** list) { /* Plotting list to stdout */
+	printf ("->plot<-\n");
+	if (!list)
+		return;
+	--list;
+	while (*list != 0) {
+		list = (char**)*list;
+		printf ("*item:%s", *list);
+		++list;
+	}
+}
+void StringListRemove(char** list, char* str) {/* Removes all occurrences of str in the list. */			
+	printf ("->Remove<-\t%s", str);
+	if (!list)
+		return;
+	--list;
+	while (*list != 0) {
+		char ** nextItem = (char **)*list;				
+		if (!strcmp(*nextItem, str)){			
+			++nextItem;
+			nextItem = (char **)* nextItem;		
+			printf ("\t\tfree\t\t%p\n", *list);
+			free(*list);			//	e
+			*list = (char *)nextItem;	//	g
+		} else {					
+			list = ++nextItem;
+		}
```

Comment:
If you free the strings inside list, you also free the exact c strings in the program. Take a look at the comments in the StringListAdd function